swiggy-pdp-cart/src/content.js

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import Menu from "./menu";
+import Cart from "./cart";
+import Categories from "./categories";
+
+export default class Content extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.state = {
+      category: "recommended",
+    };
+  }
+
+  createMenuList(category) {
+    const menu = fetchMenuList();
+    const menuList = menu.filter((dish) => dish.categories.includes(category));
+    return menuList;
+  }
+
+  handleChange(category) {
+    this.setState({
+      category: category,
+    });
+  }
+  render() {
+    const categoryList = fetchCategoriesList();
```

Comment:
`fetchCategoryList` - this will cause the data to be fetched every time the component is rendered - we need to make the API call only once when the component mounts - move the API call to `componentDIdMount`